test/fhirtypes/applyInsertRules.test.ts

```diff
@@ -0,0 +1,302 @@
+import { FSHTank, FSHDocument } from '../../src/import';
+import { Profile, RuleSet, FshCode, FshValueSet } from '../../src/fshtypes';
+import { InsertRule, CardRule, ConceptRule } from '../../src/fshtypes/rules';
+import { loggerSpy, assertCardRule, assertValueSetConceptComponent } from '../testhelpers';
+import { minimalConfig } from '../utils/minimalConfig';
+import { applyInsertRules } from '../../src/fhirtypes/common';
+
+describe('applyInsertRules', () => {
```

Comment:
Could you add tests to demonstrate the scenarios where a `RuleSet` is inserted multiple times? I'm thinking of two main cases: one where the `RuleSet` is simply repeated, and one where the it's less direct:
```
RuleSet: Cake
* ...

RuleSet: Pie
* insert Cake
* ...

Profile: BakeryProfile
* insert Cake
* insert Pie
```

Reply:
Thanks for pointing this out.  The case you laid out was erroneously being logged as a circular dependency!